src/data/articles/2023/09/01/rendering-modes.md

```
@@ -0,0 +1,61 @@
+
+Astro pages, routes, and API endpoints can be either [pre-rendered at build time](#pre-rendered) or [rendered on demand by a server](#on-demand-rendered) when a route is requested. With [Astro islands](/en/concepts/islands/), you can also include some client-side rendering when necessary.
+
+In Astro, most of the processing occurs on the server, instead of in the browser. This generally makes your site or app faster than client-side rendering when viewed on less-powerful devices or on slower internet connections. Server-rendered HTML is fast, SEO friendly, and accessible by default.
+
+## Server `output` modes
+
+You can configure how your pages are rendered in your [`output` configuration](/en/reference/configuration-reference/#output).
+
+### Pre-rendered
+
+The **default rendering mode is **`output: 'static'`\*\*\*\*, which creates the HTML for all your page routes at build time.
```

markdownlint-cli2 (0.17.2)

Spaces inside emphasis markers
null

(MD037, no-space-in-emphasis)